segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -229,10 +230,23 @@ public void commitRecords(Collection<SystemJournalRecord> records) throws ChunkS
         }
         // Persist
         synchronized (lock) {
-            writeToJournal(bytes);
-            // Add a new log file if required.
-            if (!chunkStorage.supportsAppend() || !config.isAppendEnabled()) {
-                startNewJournalFile();
+            boolean done = false;
+            while (!done) {
```

Comment:
Can you do this with the `Retry` class (in `common`). We have various retry utilities that can help with your syntax here.

Reply:
I need to retry only if specific (wrapped) exception was thrown. I don't think Retry can do that.

Reply:
`.retryWhen(ex -> Exceptions.unwrap(ex) instanceof xyz)`